### ISSUE TITLE
Possible Bug in AES.MODE_CFB buffer update?

### ISSUE THREAD

**Issue by arunanshub** [Legrandin/pycryptodome#420]
Here is the code to demonstrate that:
```python
from Crypto.Cipher import AES
import os

k, n = os.urandom(32), os.urandom(16)
buf = bytearray(10)

x = AES.new(k, AES.MODE_CFB, n)
# update the buffer here
x.encrypt(buf, buf)

x = AES.new(k, AES.MODE_CFB, n)
# update the buffer: decrypt
x.decrypt(buf, buf)
```

The `buf` must be a bytearray of Null, but that is not the case.

![Screenshot_20200525-223323](https://user-images.githubusercontent.com/48434243/82832077-c34d4600-9ed7-11ea-998f-884e98af69fa.png)

Although, the non-buffer decryption works normally.

```python
k, n = os.urandom(32), os.urandom(16)
buf = bytearray(10)

x = AES.new(k, AES.MODE_CFB, n)
# update the buffer here
x.encrypt(buf, buf)

x = AES.new(k, AES.MODE_CFB, n)
# update the buffer: decrypt
x.decrypt(buf)  # returns Null byte of length 10
```

![Screenshot_20200525-230410~2](https://user-images.githubusercontent.com/48434243/82833665-4a042200-9edc-11ea-8629-1bb19b1a70d2.png)

**The bug seems to be in the `decrypt` method's buffer operation.**

**Comment by texadactyl** [comment]
@Legrandin  I believe that this is a bug.  Decryption is not setting the output correctly when the output buffer is passed as a second argument.  Note that this is not an issue in CBC or OFB modes.  Probably specific to CFB.
```
"""
Issue #420
"""
from Crypto.Cipher import AES
import os

def test1():
    """
    Encryption and decryption both return output
    """
    key, iv = os.urandom(32), os.urandom(16)
    buf1 = buf2 = bytearray(16)
    print("Cleartext 1:", buf1)

    cipher1 = AES.new(key, AES.MODE_CFB, iv)
    buf2 = cipher1.encrypt(buf1)
    print("Ciphertext:", buf2)

    cipher2 = AES.new(key, AES.MODE_CFB, iv)
    buf1 = cipher2.decrypt(buf2)
    print("Cleartext 2: ", buf1)

def test2():
    """
    Encryption uses an output buffer argument.
    Decryption returns output.
    """
    key, iv = os.urandom(32), os.urandom(16)
    buf1 = buf2 = bytearray(16)
    print("Cleartext 1:", buf1)

    cipher1 = AES.new(key, AES.MODE_CFB, iv)
    cipher1.encrypt(buf1, buf2)
    print("Ciphertext:", buf2)

    cipher2 = AES.new(key, AES.MODE_CFB, iv)
    buf1 = cipher2.decrypt(buf2)
    print("Cleartext 2: ", buf1)

def test3():
    """
    Both use an output buffer argument.
    """
    key, iv = os.urandom(32), os.urandom(16)
    buf1 = buf2 = bytearray(16)
    print("Cleartext 1:", buf1)

    cipher1 = AES.new(key, AES.MODE_CFB, iv)
    cipher1.encrypt(buf1, buf2)
    print("Ciphertext:", buf2)

    cipher2 = AES.new(key, AES.MODE_CFB, iv)
    cipher2.decrypt(buf2, buf1)
    print("Cleartext 2: ", buf1)

test3()

```
In the above program, test1() and test2() execute correctly but test3() does not.

**Comment by Legrandin** [comment]
Fix available in commit a1a9791ae071f67f49b6176f7f43de648d1838c0.

**Comment by frispete** [comment]
Impressive performance @Legrandin.

**Comment by Legrandin** [comment]
The fix is now in v3.9.8
